### DISCUSSION TITLE
How to add the Agent and rejectUnauthorized into pactum?

### DISCUSSION THREAD

**Discussion by kvernon** [pactumjs/pactum#282] [Q&A] [answered]
Hi all,

We're trying out pactumjs, and I'm stuck on how to test a local server. Currently I get a failure of `CERT_HAS_EXPIRED`

```text
console. Error
    Error performing request

  ... 

  console. Error
    {
      "code": "CERT_HAS_EXPIRED"
    }

```

Now, that's expected, we don't care internally that the cert is invalid, and in other libraries; including nodejs, we could pass in the agent to help bypass this and get the data.

```javascript
import { Agent } from 'https';

const agent = new Agent({
    rejectUnauthorized: false
});
```

How do I apply this, or the `rejectUnauthorized=false` into pactumjs?

When trying, I honestly cannot find something that relates to it:

```typescript
    //...
    await spec()
        .post(config.url)
        .withHeaders({
            'world': config.world,
            'Content-Type': 'application/json',
        })
        .withJson({
            "destination": "createSession",
            "params": {
                "world_id": 2081
            }
        })
        .expectStatus(200);

```

**Comment by cbruchwaldnetfonds** [comment, marked as answer]
Have you checked withCore? https://pactumjs.github.io/api/requests/withCore.html

**Comment by kvernon** [comment]
Holy crap! That was it perfectly!

Thank you,

Kelly
